### ISSUE TITLE
Extract seems to stop working when unpacking a ZIP file uploaded from a Windows machine

### ISSUE THREAD

**Issue by kurbar** [ZJONSSON/node-unzipper#175]
We noticed some interesting behaviour regarding the `Extract` method.
I had created a testing ZIP file on a MacOS system and tried uploading it to our system. ZIP is identified with a mimetype of `application/zip` and `Extract` has no problem unpacking it if the file originated from a MacOS or Linux machine.

However, if the same ZIP file was uploaded from a Windows machine, then the mimetype gets identified as `application/x-zip-compressed` and `Extract` seems to just get stuck or stop. No output is given or anything and the rest of the code execution stops, doesn't seem to even reach the catch block.

```
try {
  await fs.createReadStream(filePath)
    .pipe(unzipper.Extract({ path: unpackDir }))
    .promise();
  console.log('ZIP file decompressed. Deleting archive file as no longer needed'); // This never gets called
} catch (err) {
  console.error(err); // This never gets called as well
}
```

Can't figure out if it has anything to do with the mimetype or anything.

**Comment by kurbar** [comment]
Using the fixes from PR #172 fixed the issue for me.

**Comment by ZJONSSON** [comment]
Thanks @kurbar - very helpful.   #172 is merged, please reopen if any further issues.
